src/containers/Pensjonsberegning/Pensjonsberegning.tsx

```diff
@@ -19,9 +24,33 @@ export function Pensjonsberegning() {
     isError,
     isSuccess,
   } = useTidligsteUttaksalderQuery()
-
+  const harSamtykket = useAppSelector(selectSamtykke)
   const [valgtUttaksalder, setValgtUttaksalder] = useState<string | undefined>()
 
+  const {
+    data: pensjonsavtaler,
+    isLoading: isPensjonsavtalerLoading,
+    isError: isPensjonsavtalerError,
+  } = usePensjonsavtalerQuery(
```

Comment:
Denne hentes to nivåer over der den brukes, noe som er lite hensiktsmessig og fort blir rotete. Vi burde kalle denne direkte i `Pensjonsavtaler`-komponenten

Reply:
Det er to grunn til at jeg har strukturert det på denne måten:
- Responsen skal brukes både i Grunnlag og i Pensjonssimulering.
- Komponenter skal være dumbs og håndtere så lite logikk som mulig (bare visningen), mens logikken er abstrahert i container. Det er poenget med redux containers. De skal holder logikken slik at koden er decoupled. Du kan lese mer om Redux og containers f.eks her: https://blog.jakoblind.no/are-container-components-necessary-when-using-redux/

Reply:
Da foreslår jeg at vi henter dataen begge steder der det brukes, både i `Grunnlag` og `Pensjonssimulering`. Skal `Grunnlag` ha props som `showLoaderForPensjonsavtaler`, `showErrorForPensjonsavtaler`, `showLoaderForAlderspensjon`, `showErrorForAlderspensjon`, `showLoaderForSivilstand`, `showErrorForSivilstand`, etc? Det blir jo helt håpløst rotete i lengden. 

Du har dessuten en pr ute nå hvor du bruker `getPersonQuery` i Sivilstand, som er en komponent som brukes i `Grunnlag`. Skal ikke denne være en dum komponent om du skal gjøre som du selv sier her? Skal vi kalle `getPersonQuery` i `Pensjonsberegning` og bare sende det videre ned 3-4 nivåer til der det faktisk er interessant å bruke den?

Reply:
`usePensjonstavtalerQuery` er en tyngre push request som tar imot noen parametre. Det blir duplisert kode i søskenkomponenter om vi putter denne logikken i begge og er grunnen til at jeg ikke har gjort det. `Grunnlag` og `Pensjonssimulering` har samme parent, nettopp en container som kan håndtere logikken for dem. Det er dermed ikke nødvendig å forurense disse enkle komponenter med dobbelt opp logikk.
Dataen som brukes i `Sivilstand` derimot er unik til sivilstand, og er mye enklere. Det er grunnen til at jeg ikke så behov for å innføre noe container der.
Så nei, container skal ikke være systematisk, men er en metode for å holde logikken adskilt der det er behov.

Reply:
Du trenger ikke sende med samme parametre begge steder: https://redux-toolkit.js.org/rtk-query/usage/queries#selecting-data-from-a-query-result

Reply:
Det må komme fra parenten. Jeg kan foreslå å beholde containeren og bruke `selectFromResult` i begge barna isåfall

Reply:
Mener du at en parent-komponent må kalle query-hooken? Er du sikker på det? Tror ikke det skal ha noe å si siden resultatet uansett havner i den globale storen. Men jeg liker forslaget, syns det er ryddig at selve kallet foregår i `Pensjonsberegning` og at man henter ut dataen der den brukes med `selectFromResult` 👍

Reply:
Både parent og child må kalle query-hooken. Og ved å lese dokumentasjonen ser jeg at parametrene må sendes begge steder. Jeg har refaktorert slik at parametrene sendes ned til barna istedenfor loading/error properties. Da kaller de samme query-hook og får data fra cachem.
Jeg foretrekker nok slik jeg hadde først skrevet, men har ikke noe sterk mening. Så hvis du er fornøyd med den nye versjonen er det greit for meg.